File: pjecalc-fonte/negocio/br/jus/trt8/pjecalc/negocio/dominio/cartaodeponto/FiltroDoCartaoDePonto.java
/*
 * Decompiled with CFR 0.152.
 * 
 * Could not load the following classes:
 *  org.hibernate.Criteria
 *  org.jboss.seam.annotations.Name
 */
package br.jus.trt8.pjecalc.negocio.dominio.cartaodeponto;

import br.jus.trt8.pjecalc.base.comum.CriteriosDePesquisa;
import br.jus.trt8.pjecalc.base.comum.FiltroBase;
import br.jus.trt8.pjecalc.negocio.dominio.calculo.Calculo;
import br.jus.trt8.pjecalc.negocio.dominio.cartaodeponto.CartaoDePonto;
import br.jus.trt8.pjecalc.negocio.dominio.cartaodeponto.RepositorioDeCartaoDePonto;
import java.io.Serializable;
import java.util.List;
import org.hibernate.Criteria;
import org.jboss.seam.annotations.Name;

@Name(value="filtroDoCartaoDePonto")
public class FiltroDoCartaoDePonto
extends FiltroBase<CartaoDePonto>
implements Serializable {
    private static final long serialVersionUID = -7235727704431456701L;
    private String nome;
    private Calculo calculo;

    public FiltroDoCartaoDePonto() {
        super(RepositorioDeCartaoDePonto.class);
    }

    @Override
    protected void elaborarCriterio(Criteria criteria) {
    }

    @Override
    public List<CartaoDePonto> filtrar() {
        CriteriosDePesquisa criterios = new CriteriosDePesquisa("id asc");
        if (this.nome != null && !this.nome.equals("")) {
            criterios.adicionarCriterio("and", "nome like ?", "%" + this.nome.toUpperCase() + "%");
        }
        if (this.calculo != null && this.calculo.getId() != null) {
            criterios.adicionarCriterio("and", "calculo = ?", this.calculo);
        }
        return ((RepositorioDeCartaoDePonto)this.getRepositorio()).pesquisar(criterios.getOrderBy(), criterios.getClausulaWhere().toString(), criterios.getParametros().toArray());
    }

    public String getNome() {
        return this.nome;
    }

    public void setNome(String nome) {
        this.nome = nome;
    }

    public Calculo getCalculo() {
        return this.calculo;
    }

    public void setCalculo(Calculo calculo) {
        this.calculo = calculo;
    }
}

File: src/lib/pjecalc/core/dominio/cartaodeponto/filtro-do-cartao-de-ponto.ts
/**
 * PJe-Calc v2.15.1 — FiltroDoCartaoDePonto (stub)
 * Porte de: br.jus.trt8.pjecalc.negocio.dominio.cartaodeponto.FiltroDoCartaoDePonto
 */
import type { Calculo } from '../calculo/calculo';
import type { CartaoDePonto } from './cartao-de-ponto';

export class FiltroDoCartaoDePonto {
  private calculo: Calculo | null = null;
  private nome: string | null = null;

  getCalculo(): Calculo | null { return this.calculo; }
  setCalculo(c: Calculo | null): void { this.calculo = c; }

  getNome(): string | null { return this.nome; }
  setNome(v: string | null): void { this.nome = v; }

  filtrar(): CartaoDePonto[] {
    return []; // TODO(fase-13/infra)
  }
}
